### DISCUSSION TITLE
Error creating SSL - Site is not reachable

### DISCUSSION THREAD

**Discussion by carloscds** [rcl-ssl/docs#6] [General]
ERROR: Could not reach your site at http://xxx.xxx.xxx. Please ensure your site is reachable at this domain and try again in 10 minutes. Closing and re-opening this site or clearing your browser cache may help.

But If I check DNS it's working

**Comment by carloscds** [comment]
Nobody ?

**Reply by rcladmin** [reply]
what the the domain of the site?

**Reply by carloscds** [reply]
api.kwing.com.br

**Reply by rcladmin** [reply]
The 'HEAD' method is not allowed in your API. The RCL app uses a 'HEAD' request to determine if a site exists, not a 'GET' request.. Please allow a 'HEAD' request in your API.

**Reply by carloscds** [reply]
Yes, it's default for APIs, why your software need this approach ?

**Reply by rcladmin** [reply]
HEAD requests are normally used to check the existence of a website. It has lower overhead since the body of the website is not returned. Only the head is returned. Almost all websites allow HEAD requests.

Since you are working with an API, the HEAD request may be not allowed in your API. 

If you a using an aspnet/core Web API, you can decorate your controller  with the [HttpHead] attribute to allow the HEAD request for route: api.kwing.com.br.

You can test you API in Postman as shown below:

<img width="631" alt="Capture" src="https://user-images.githubusercontent.com/4339259/118998966-e0d0b680-b957-11eb-8670-81cb64ce35dc.PNG">

**Reply by carloscds** [reply]
I don't have controller at / because it's an API.

**Reply by rcladmin** [reply]
what technology is the API built with? APNET Web API? PHP? Java?

**Reply by rcladmin** [reply]
I am provided a fix so that if the site does not support 'HEAD' , a 'GET' request will be used instead. You can retry the app is the next 30 mins.

**Reply by rcladmin** [reply]
Ok, you can retry creating a TLS/SSL certificate now for your web app

**Reply by carloscds** [reply]
Now I have this error:
Error:Failed to create app service certificate for host : api.kwing.com.br on siteId : 824 with challenge : HTTP for challenge : HTTP. Could not validate api.kwing.com.br with challenge http-01 for domain api.kwing.com.br, status reported : invalid. Invalid response from http://api.kwing.com.br/.well-known/acme-challenge/nK10vz2wXaqwDopdYje4gS2NjUaw0HsDutzfR8jMK3c [20.49.97.15]: 404 . Recommendation R03: You did not correctly add the validation file to the root of your website, please check that your validation record is available publicy on the web before you validate.. Help link: https://docs.rclapp.com/portal/stand-alone.html#create-a-ssltls-certificate-using-http-01
Information:Waiting for process to complete ****

**Reply by rcladmin** [reply]
You API is not hosting the validation file publicly. See image below:

<img width="648" alt="Capture" src="https://user-images.githubusercontent.com/4339259/119035635-c7426580-b97d-11eb-8d87-4be1c57a0a24.PNG">

To use the HTTP challenge, you will need to ensure that the content at the route : 

http://api.kwing.com.br/.well-known/acme-challenge/ can be accessed by the Let's Encrypt Server. 

What technology are you using to build your API. If you provide this information , I may be able to advise you better.

**Reply by carloscds** [reply]
ASP.NET Core 5.0

**Reply by rcladmin** [reply]
This may help you:

First we need to check if the RCL app is adding the necessary validation files to your app service.

In the Azure Portal for your web app, open your app service in the App Service Editor. in the root of your website you should see the following folders and file as shown below. The validation file will be named differently.

<img width="775" alt="Capture" src="https://user-images.githubusercontent.com/4339259/119038868-a5e37880-b981-11eb-974c-3f188510c7f9.PNG">

Please look in the wwwroot folder for these folders and file. Let me know if you find the folders and file in your app service.

**Reply by carloscds** [reply]
Done and fixed! Tks.
